sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -41,36 +42,62 @@
     private SetStatementAnalyzer() {
     }
 
-    public static SetAnalyzedStatement analyze(SetStatement node, ParameterContext parameterContext) {
-        Settings.Builder builder = Settings.builder();
+    public static SetAnalyzedStatement analyze(SetStatement node, Analysis context) {
+        boolean isPersistent = node.settingType().equals(SetStatement.SettingType.PERSISTENT);
+        Row parameters = context.parameterContext().parameters();
 
-        if (!SetStatement.Scope.GLOBAL.equals(node.scope())) {
-            logger.warn("SET STATEMENT WITH SESSION OR LOCAL WILL BE IGNORED: {}", node);
+        if (SetStatement.Scope.SESSION.equals(node.scope())) {
             Assignment assignment = node.assignments().get(0);
-            String settingsName = ExpressionToStringVisitor.convert(assignment.columnName(),
-                parameterContext.parameters());
-            Set<String> settingNames = CrateSettings.settingNamesByPrefix(settingsName);
-            if (settingNames.size() != 0) {
-                throw new IllegalArgumentException(String.format(Locale.ENGLISH, "GLOBAL Cluster setting '%s' cannot be used with SET SESSION / LOCAL", settingsName));
+            String setting = ExpressionToStringVisitor.convert(assignment.columnName(), parameters);
+            verifySessionOrLocalSetting(setting);
+
+            Settings.Builder builder = Settings.builder().put(setting, assignment.expressions());
+            // The search_path takes a schema name as a string or comma-separated list of schema names.
+            // In the second case only the first schema in the list will be used.
+            if (setting.equals("search_path")) {
+                Expression value = assignment.expression();
+                if (value != null) {
+                    String searchPath = ExpressionToStringVisitor.convert(value, parameters);
+                    String schemas[] = searchPath.split(",");
```

Comment:
couldn't really find smth more convenient in our code base. Improved it a bit by getting rid of the `trim` call.